src/methods/clipping/clipping_processor.jl

```diff
@@ -0,0 +1,286 @@
+    status = false
+    for ii in eachindex(b_list)
+        if ii == 1
+            temp = within(edges_b[ii][1], poly_a)
+            status = !(temp[1])
+            continue
+        end
+        if b_list[ii].inter
+            b_list[ii].ent_exit = status
+            status = !status
+        end
+    end
+
+    return a_list, b_list
+end
+
+"""
+    _build_ab_list(
+        poly_a, poly_b
+    )::a_list, b_list, a_idx_list, intr_list, edges_a, edges_b
+
+    This function calls '_build_a_list', '_build_b_list', and '_flag_ent_exit'
+    in order to fully form a_list and b_list. The 'a_list' and 'b_list'
+    that it returns are the fully updated vectors of PolyNodes that represent
+    'poly_a' and 'poly_b', respectively. This function also returns
+    'a_idx_list', which at its "ith" index stores the index in 'a_list' at
+    which the "ith" intersection point lies. This function also returns
+    'intr_list', which at the "ith" index contains the cartesian coordinates
+    for the "ith" intersection point. Finally, this function returns 'edges_a'
+    and 'edges_b' which are vectors of the cartesian points of the original
+    polygons, 'poly_a' and 'poly_b'.
+"""
+function _build_ab_list(poly_a, poly_b)
+    # Make a list for nodes of each polygon. Note the definition of PolyNode
+    a_list = Array{PolyNode, 1}(undef, _nedge(poly_a))
+    b_list = Array{PolyNode, 1}(undef, _nedge(poly_b))
+
+    # Initialize arrays to keep track of the important information 
+    # associated with the intersection points of poly_a and poly_b.
+    # I initialize these arrays assumed there will be a maximum of 
+    # 30 intersection points and then I truncate the arrays later.
+    k = 4
+    # intr_list stores the cartesian coordintes of the intersection point
+    intr_list = Array{Tuple{Real, Real}, 1}(undef, k)
+    # At index i, a_idx_list stores the index of the ith intersection point in a_list
+    a_idx_list = Array{Int, 1}(undef, k)
+    # At index i, b_idx_list stores the edge number of poly_b that the ith intersection
+    # point lies on.
+    b_idx_list = Array{Int, 1}(undef, k)
+    # Alpha values are used to determine the order in which to place
+    # intersection points in a polygon list, which is especially
+    # useful when many intersection points lie on the same edge of a polygon.
+    alpha_a_list = Array{Real, 1}(undef, k)
+    alpha_b_list = Array{Real, 1}(undef, k)
+
+    # These lists store the cartesian coordinates of poly_a and poly_b
+    edges_a = to_edges(poly_a)
+    edges_b = to_edges(poly_b)
+
+    intr_list, a_idx_list, b_idx_list, 
+    alpha_a_list, alpha_b_list = _build_a_list(edges_a, edges_b, intr_list, a_idx_list, b_idx_list, 
+                                                        alpha_a_list, alpha_b_list, a_list, b_list)
+    b_neighbors, b_list = _build_b_list(b_list, intr_list, a_idx_list, b_idx_list, alpha_b_list)
+
+    # Iterate through a_list and update the neighbor indices
```

Comment:
Should be able to get rid of this loop if we update A neighbors as we go as described above.